drivers/storage/portworx/util/util.go

```diff
@@ -1151,16 +1153,25 @@ func AuthEnabled(spec *corev1.StorageClusterSpec) bool {
 }
 
 // SetupContextWithToken Gets token or from secret for authenticating with the SDK server
-func SetupContextWithToken(ctx context.Context, cluster *corev1.StorageCluster, k8sClient client.Client) (context.Context, error) {
+func SetupContextWithToken(ctx context.Context, cluster *corev1.StorageCluster, k8sClient client.Client, skipSecurityCheck bool) (context.Context, error) {
```

Comment:
The extra param  added to bypass the check is because there are three scenarions during the process:
1. Security disabled on STC and nodes
2.  Security enabled on STC and nodes
3.  Security disabled on STC and enabled on nodes, because node update is not yet done
4.  Security enabled on STC and disabled on nodes, because node update is not yet done

The scenario specific to this PR will be as follow #3, 

get secret
if err != nil && err != NotFound {
    return err
}
if err == NotFound {
    if AuthEnabled {
        // this is scenario #2
        // we must have the secret since auth is enabled in stc
        // even if the calls to the nodes not upgraded yet might work without a secret this is not a valid config
        return error
    }
    **// this could be scenario #1 and #3**
    **// #1 In normal scenario when AuthEnabled is false , this is ok**
    **/ /#3  but in case when AuthEnabled in STC is disabled, but nodes are not upgraded, token is expected**
    // This is ok. There is no token to add anyway so we should be good.
    return ctx, nil
}
// secret is found
// add token from the secret to the context and continue. It does not matter whether auth is enabled or not.
// This will work whether the node has been upgraded or not.

Reply:
> / / 3 but in case when AuthEnabled in STC is disabled, but nodes are not upgraded, token is expected

In this case, the token is not found. So, no valid token can be supplied anyway. Yes?